File: java/net/mikekohn/java_grinder/IceFun.java
/**
 *  Java Grinder
 *  Author: Michael Kohn
 *   Email: mike@mikekohn.net
 *     Web: https://www.mikekohn.net/
 * License: GPL
 *
 * Copyright 2014-2023 by Michael Kohn
 *
 */

package net.mikekohn.java_grinder;

public class IceFun
{
  private IceFun()
  {
  }

  /** Set musical tone. Valid values are 60 to 96 (C4 to C7). Value
      of 0 will turn the speaker off. */
  public static void setTone(int value) { }

  /** Change value of servo peripheral. Valid values are 12000 to 24000
      where 12000 is full rotation in one direction and 24000 is full
      rotation in the reverse direction. */
  public static void setServo(int index, int value) { }

}

File: api/ice_fun.cxx
/**
 *  Java Grinder
 *  Author: Michael Kohn
 *   Email: mike@mikekohn.net
 *     Web: https://www.mikekohn.net/
 * License: GPLv3
 *
 * Copyright 2014-2018 by Michael Kohn
 *
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdint.h>

#include "api/ice_fun.h"

#define CHECK_FUNC(funct,sig) \
  if (strcmp(#funct#sig, method_name) == 0) \
  { \
    return generator->ice_fun_##funct##sig(); \
  }

#define CHECK_FUNC_CONST(funct,sig) \
  if (strcmp(#funct#sig, method_name) == 0) \
  { \
    return generator->ice_fun_##funct##sig(const_val); \
  }

#define CHECK_FUNC_CONST_2(funct,sig) \
  if (strcmp(#funct#sig, function) == 0) \
  { \
    return generator->ice_fun_##funct##sig(const_val1, const_val2); \
  }

int ice_fun(
  JavaClass *java_class,
  Generator *generator,
  const char *method_name)
{
  CHECK_FUNC(setTone,_I)
  CHECK_FUNC(setServo,_II)

  return -1;
}

int ice_fun(
  JavaClass *java_class,
  Generator *generator,
  const char *method_name,
  int const_val)
{
  return -1;
}

int ice_fun(
  JavaClass *java_class,
  Generator *generator,
  const char *function,
  int const_val1,
  int const_val2)
{
  return -1;
}

